### PR TITLE
agent-client-js: add withRefs interface

### PR DESCRIPTION
**Pull request by simonvadee**
`(process|segment).withRefs(references)` adds a `refs` field containing the list `references`, which are then passed to the agent through the adaptor.

---
This change is [Reviewable](https://reviewable.io/reviews/stratumn/indigo-js/84)

### CODE REVIEW

File: packages/agent-client-js/test/createMap.js [hunk 1]
```diff
@@ -25,6 +25,20 @@ describe('#createMap', () => {
           segment.link.state.title.should.be.exactly('Test');
         }));
 
+    it('creates a map with references', () =>
```
Comment (alexppxela) [inline packages/agent-client-js/test/createMap.js]: Is it possible to add a test with a bad reference (missing linkHash for example)?

File: packages/agent-client-js/src/segmentify.js [hunk 2]
```diff
@@ -14,13 +14,21 @@
   limitations under the License.
 */
 
+import withRefs from './withRefs';
+
 export default function segmentify(adaptor, process, obj) {
   Object.keys(process.processInfo.actions)
-    .filter(key => ['init'].indexOf(key) < 0)
```
Comment (such) [inline packages/agent-client-js/src/segmentify.js]: good call to change the name of the variable

**Comment by simonvadee** [comment]
Review status: 0 of 9 files reviewed at latest revision, 1 unresolved discussion.

---

*[packages/agent-client-js/test/createMap.js, line 28 at r1](https://reviewable.io:443/reviews/stratumn/indigo-js/84#-KxnFFayO2riHvSoAOTH-r1-28:-Kxw4eSEYhXstUdD0Ykn:b-896fix) ([raw file](https://github.com/stratumn/indigo-js/blob/e3029ffbb7ec12656b32d64c5700a6d9702047cb/packages/agent-client-js/test/createMap.js#L28)):*
<details><summary><i>Previously, alexppxela (Alexandre Thibault) wrote…</i></summary><blockquote>

Is it possible to add a test with a bad reference (missing linkHash for example)?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/stratumn/indigo-js/84)*

**Comment by alexppxela** [comment]
Reviewed 7 of 8 files at r1, 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/stratumn/indigo-js/84)*

**Pull request merged**